database/gormdb_assignm_test.go

```diff
@@ -117,62 +117,46 @@ func TestGetNextAssignment(t *testing.T) {
 		t.Errorf("expected unapproved assignment to be %v, got %v", assignment1.ID, nxtUnapproved.ID)
 	}
 
-	// approve submission1
-	if err := db.UpdateSubmission(submission1.ID, true); err != nil {
+	// approve submission for assignment1
+	if err := db.UpdateSubmission(1, true); err != nil {
```

Comment:
Why change this from `submission1.ID` to just `1`?

Reply:
Or rather, why not use `assignment1.ID`?

Reply:
`submission1.ID` is 0 here. `assignment1.ID` will work here, but not on the real database. Is this allowed?

Reply:
Sorry. I'm got confused by the comment referring to `assignment1`. It is a `submissionID`, so we should not use `assignment1.ID` here; you are right.
However, if `submission1.ID` does not work, we should probably make sure that `submission1.ID` is not 0.

Reply:
But it seems like it is the database that is generating the `Submission.ID`... using the `CreateSubmission` method.

Reply:
It does have ID in the database, but locally the struct is `submission1 := pb.Submission{AssignmentID: assignment1.ID, UserID: user.ID}`, so, no ID.

Reply:
Actually, I've fixed it: gorm method will populate the query structs just as with assignments, so that IDs can be used directly.

Reply:
I thought so... If you can resolve the conflict; I'll approve this PR and we can start another PR to fix the database methods (see Trello card).

Reply:
Done.